src/components/SignInPageForm/index.js

```diff
@@ -10,6 +16,22 @@ class Form extends React.Component {
         // Password Managers need these attributes to be able to identify the form elements properly.
         this.form.setAttribute('method', 'post');
         this.form.setAttribute('action', '/');
+
+        this.form.addEventListener('submit', this.preventFormDefault);
+    }
+
+    componentWillUnmount() {
+        if (!this.form) {
+            return;
+        }
+
+        this.form.removeEventListener('submit', this.preventFormDefault);
+    }
+
+    preventFormDefault(event) {
+        // When enter is pressed form is submitted to action url (POST /).
+        // As we are using controlled component, we need to disable it here.
+        event.preventDefault();
     }
```

Comment:
This is intentional right, why prevent that?

cc: @thienlnam

Reply:
Yeah I believe so, if this was using the form default submit action before then how does this work now? 

Is there a onSubmit handler for the form now? Also if there is, then you can prevent eventDefault via the onSubmit handler instead of adding an event listener like this

Reply:
https://github.com/Expensify/App/blob/94057baaf27d1f6c28277fc5cd713ee42ec10ada/src/pages/SetPasswordPage.js#L103

Form submit is handled in this component. I believe, there is no `onSubmit` prop in `react-native-web`, this is the reason handled in this way.

Reply:
We do not use the [Form](https://github.com/Expensify/App/blob/main/src/components/Form.js) here, curious why?

@luacmartins or @thienlnam ?

Reply:
I think mostly because Form doesn't have a couple of features used in the sign in pages:

1. Auto-submit the form when entering a 6-digit 2FA code
2. This dark magic ([here](https://github.com/Expensify/App/blob/60b9d02cab9511b21da32ab2bc9e3ccc47f2be2e/src/pages/signin/SignInPage.js#L80-L81) and [here](https://github.com/Expensify/App/blob/60b9d02cab9511b21da32ab2bc9e3ccc47f2be2e/src/pages/signin/LoginForm.js#L199-L200)) to keep both `LoginForm` and `PasswordForm` mounted, but visibly hidden